### PR TITLE
Add error handling on Preview page

### LINKED ISSUE #315: Join Section | Remove unused variables | Add Constants  | Final Error handling
# Issues
- Remove unused variables
- Add constants to help DRY
- add error handling on the preview page

- [ ] Remove Unused Variables
  - The code contains unused variables for the previous button.
  - Files to be changed:
    - ```join.js```
- [ ] Add Constants To Help DRY
  - To ensure DRY we will be adding a few constants to a file.
  - Currently Required constants:
    - ``` Base URL = https://api.realdevsquad.com/```
    - ```Login URL = <github_login_url>```
    - ```self URL = <api_self_call_url>``` 
  - Files to be changed :
    - ```constants.js```
    - ```join.js```
- [ ] Add error Handling on the Preview Page
  - Currently, if the user manages to get to the preview page they can directly go to the completed/final page by clicking submit. This happens irrespective of whether the data is being submitted or not. We need to add an error-handling mechanism to prevent this and give an appropriate response to the user.
  - Files to change:
    - ```join.js```

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
closes #315 
This PR includes refactoring code for removing unused variables. Creating constants to ensure DRY and handling data submissions of join data.

For mode details Please refer the issue ticket #315

### CODE REVIEW

File: js/constants.js [hunk 1]
```diff
@@ -256,3 +256,8 @@ export const MAP_COORDINATES = {
   LNG: 78.476681027237,
   LAT: 22.1991660760527,
 };
+export const BASE_URL = 'https://api.realdevsquad.com/';
+export const JOIN_POST_URL = BASE_URL + 'users/self/intro';
```
Comment (heyrandhir) [inline js/constants.js:260]: Nit: we can use ES6 template literals 
```suggestion
export const JOIN_POST_URL = `${BASE_URL}/users/self/intro`;
```

File: js/constants.js [hunk 2]
```diff
@@ -256,3 +256,8 @@ export const MAP_COORDINATES = {
...
+export const SELF_URL = BASE_URL + 'users/self';
+export const LOGIN_URL =
+  'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97';
```
Comment (heyrandhir) [inline js/constants.js:263]: Is passing the client id mandatory here ?

File: js/join.js [hunk 3]
```diff
@@ -111,10 +112,9 @@ function fetchSavedDetails() {
       window.localStorage.setItem('lastName', res.last_name);
       if (res.statusCode === 401) {
         alert('You are not logged in! Redirecting you to login.');
-        location.href =
-          'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97';
+        location.href = LOGIN_URL;
       }
-      url = `https://api.realdevsquad.com/users/${res.id}/intro`;
+      url = BASE_URL + 'users/' + res.id + '/intro';
```
Comment (heyrandhir) [inline js/join.js:117]: Nit: We can use ES6 template literal.

**Review by heyrandhir — APPROVED** [review]

**Comment by heyrandhir** [comment]
LGTM

File: js/constants.js (hunk 2 above)
Comment (ivinayakg) [inline js/constants.js:263]: please this from this PR I'm raising another one to fix the issue and do the changes in other files requiring this variable.

Comment (RitikJaiswal75) [reply js/constants.js:263]: yes thats the URL we are using.

Comment (RitikJaiswal75) [reply js/constants.js:263]: I could not understand

**Review by heyrandhir — APPROVED** [review]
LGTM 🙂👍🏻

**Review by heyrandhir — APPROVED** [review]

**Review by heyrandhir — DISMISSED** [review]

File: join.html [hunk 4]
```diff
@@ -426,7 +426,6 @@ <h2 class="progress">Here is your personalized link</h2>
     <script src="js/navbar.js"></script>
     <script src="js/scrollBtn.js"></script>
     <script src="js/darkmode.js"></script>
-    <script src="js/featureFlag.js"></script>
```
Comment (RitikJaiswal75) [inline join.html:429]: this script was not being used in this page

**Pull request merged**